src/stim/io/measure_record_reader.cc

```diff
@@ -201,8 +225,16 @@ bool MeasureRecordReaderFormatB8::read_bit() {
 }
 
 bool MeasureRecordReaderFormatB8::next_record() {
+    while (!is_end_of_record()) {
+        read_bit();
+    }
     position = 0;
-    return false;
+    int i = fgetc(in);
+    if (i == EOF) {
+        return false;
+    }
+    ungetc(i, in);
+    return true;
```

Comment:
I thought that B8 files cannot contain more than one record, but I suppose with externally-supplied record size they can.

Reply:
All of the formats can contain an unbounded list of records.